Firestore/Source/Remote/FSTDatastore.mm

```diff
@@ -253,10 +255,9 @@ - (void)lookupDocuments:(NSArray<FSTDocumentKey *> *)keys
     [request.documentsArray addObject:[self.serializer encodedDocumentKey:key]];
   }
 
-  __block FSTMaybeDocumentDictionary *results =
-      [FSTMaybeDocumentDictionary maybeDocumentDictionary];
-
+  __block MaybeDocumentDictionary *results = nullptr;
   RPCFactory rpcFactory = ^GRPCProtoCall * {
+    results = new MaybeDocumentDictionary();
```

Comment:
Why is the initialization inside the first block? Also, if it's not a performance-critical operation, perhaps you could use `shared_ptr`?